Expensify/App
# Fix foreground notification display bug

cc @marcaaron @AndrewGable 

### Details
Turns out that we had included `@react-native-community/push-notification-ios` as a dependency in order to display the unread indicator badge on iOS. This conflicted with our UA integration, which merges foreground presentation settings from the AppDelegate. Does that even make sense? It's pretty close to being true. Details [here](https://github.com/urbanairship/react-native-module/issues/298).

### Fixed Issues
https://github.com/Expensify/Expensify/issues/144297

### Tests
1) Run the iOS app [on a physical device](https://stackoverflow.com/c/expensify/questions/6735). If you don't have an iPhone then I can find a different reviewer.
2) At the same time, run the app on web or desktop.
3) Log into one account on the iOS device, and another on the laptop.
4) With the iOS app open and in the foreground, send a message from the laptop to the iPhone. Ensure that you _do not_ receive a push notification.
5) Background the iOS app (just return to the homescreen), and send another message from the laptop to the iPhone. Ensure that you receive a push notification (don't tap the notification just yet).
6) With the iOS app still in the background, send another message from the laptop to the iPhone. Ensure that you receive a push notification. This time, tap the notification and ensure that you are taken to the relevant chat.
7) Close the iOS app, then repeat steps 1-6, replacing the iOS app with the Android app (emulator should work fine).
